### PR TITLE
feat(admin): edit automatic contribution link

### LINKED ISSUE #2303: 🚀 [Feature] [Admin Interface] Allow the admin to edit automatic contribution link
## 🚀 Feature

Allow the admin to edit an automatic contribution link.

Allow him to change:
- [ ] amount
- [ ] name
- [ ] memo
- [ ] cycle
- [ ] validFrom
- [ ] validTo
- [ ] maxAmountPerMonth
- [ ] maxPerCycle,

while keeping the original code.

The backend side is already implemented.

Consider to combine this with #2294

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
A contribution link can be changed in the admin area. 

### Issues
- fixes #2303 
 

https://user-images.githubusercontent.com/1324583/198882229-b023cedb-730a-4734-bcc4-1a361dadcc8e.mp4

### CODE REVIEW

File: admin/src/components/ContributionLinkForm.vue [hunk 1]
```diff
@@ -121,6 +127,7 @@ export default {
         return {}
       },
     },
+    editContributionLink: { type: Boolean, required: false },
```
Comment (Mogge) [inline admin/src/components/ContributionLinkForm.vue:130]: I think `required: true` is better. Else we should set `default: false`

File: admin/src/components/ContributionLinkForm.vue [hunk 2]
```diff
@@ -157,23 +164,33 @@ export default {
...
-            maxPerCycle: this.form.maxPerCycle,
-            maxAmountPerMonth: this.form.maxAmountPerMonth,
-          },
+          mutation: this.editContributionLink ? updateContributionLink : createContributionLink,
+          variables: variables,
         })
         .then((result) => {
-          this.link = result.data.createContributionLink.link
-          this.toastSuccess(this.link)
+          this.link = this.editContributionLink
```
Comment (Mogge) [inline admin/src/components/ContributionLinkForm.vue:188]: Why `this`? There is no data for link.
`const link = .... `

**Review by Mogge — CHANGES_REQUESTED** [review]

File: admin/src/components/ContributionLinkForm.vue (hunk 2 above)
Comment (ogerly) [reply admin/src/components/ContributionLinkForm.vue]: changed to const

File: admin/src/components/ContributionLink/ContributionLinkForm.vue [hunk 3]
```diff
@@ -157,23 +164,33 @@ export default {
...
+        validFrom,
+        validTo,
+        name,
+        amount,
+        memo,
+        cycle,
+        maxPerCycle,
+        maxAmountPerMonth,
+        id: this.contributionLinkData.id ? this.contributionLinkData.id : null,
+      }
```
Comment (Mogge) [inline admin/src/components/ContributionLink/ContributionLinkForm.vue:180]:
```suggestion
      const variables = {
        ...this.form,
        id: this.contributionLinkData.id ? this.contributionLinkData.id : null,
      }
```

File: admin/src/locales/de.json [hunk 4]
```diff
@@ -23,6 +24,7 @@
         "once": "einmalig"
       }
     },
+    "saveChange": "Änderung speichern",
```
Comment (Mogge) [inline admin/src/locales/de.json:27]:
```suggestion
    "saveChange": "Änderungen speichern",
```

File: admin/src/locales/de.json [hunk 5]
```diff
@@ -3,6 +3,7 @@
   "back": "zurück",
   "contributionLink": {
     "amount": "Betrag",
+    "changeSaved": "Änderung gespeichert",
```
Comment (Mogge) [inline admin/src/locales/de.json:6]:
```suggestion
    "changeSaved": "Änderungen gespeichert",
```

File: admin/src/locales/en.json [hunk 6]
```diff
@@ -3,6 +3,7 @@
   "back": "back",
   "contributionLink": {
     "amount": "Amount",
+    "changeSaved": "Change saved",
```
Comment (Mogge) [inline admin/src/locales/en.json:6]:
```suggestion
    "changeSaved": "Changes saved",
```

File: admin/src/locales/en.json [hunk 7]
```diff
@@ -23,6 +24,7 @@
         "once": "once"
       }
     },
+    "saveChange": "Save Change",
```
Comment (Mogge) [inline admin/src/locales/en.json:27]:
```suggestion
    "saveChange": "Save Changes",
```

**Review by Mogge — DISMISSED** [review]

**Comment by ulfgebhardt** [comment]
PR title should be `feat(admin): edit automatic contribution link` @ogerly

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**